src/algorithms/LG_BreadthFirstSearch_vanilla.c

```diff
@@ -0,0 +1,181 @@
+    GrB_BinaryOp second_op = (n > INT32_MAX) ? GrB_SECOND_INT64 : GrB_SECOND_INT32;
+    GrB_Semiring semiring  = NULL;
+
+    if (compute_parent)
+    {
+        // create the parent vector.  l_parent(i) is the parent id of node i
+        GrB_TRY (GrB_Vector_new(&l_parent, int_type, n)) ;
+
+        semiring = (n > INT32_MAX) ?
+            GrB_MIN_FIRST_SEMIRING_INT64 : GrB_MIN_FIRST_SEMIRING_INT32;
+
+        // create a sparse integer vector frontier, and set frontier(src) = src
+        GrB_TRY (GrB_Vector_new(&frontier, int_type, n)) ;
+        GrB_TRY (GrB_Vector_setElement(frontier, src, src)) ;
+
+        // create an index ramp (TODO: replace with apply(i) in GraphBLAS 2.0
+        GrB_Index *idx = (GrB_Index *)LAGraph_Malloc(n, sizeof(GrB_Index));
+        for (GrB_Index ix = 0; ix < n; ++ix) idx[ix] = ix;
+        GrB_TRY( GrB_Vector_new(&index_ramp, int_type, n) );
+        GrB_TRY( GrB_Vector_build(index_ramp, idx, idx, n, GrB_FIRST_UINT64) );
+        LAGraph_Free((void **) &idx);
+    }
+    else
+    {
+        // only the level is needed
+        semiring = GrB_LOR_LAND_SEMIRING_BOOL;
+
+        // create a sparse boolean vector frontier, and set frontier(src) = true
+        GrB_TRY (GrB_Vector_new(&frontier, GrB_BOOL, n)) ;
+        GrB_TRY (GrB_Vector_setElement(frontier, true, src)) ;
+    }
+
+    if (compute_level)
+    {
+        // create the level vector. v(i) is the level of node i
+        // v (src) = 0 denotes the source node
+        GrB_TRY (GrB_Vector_new(&l_level, int_type, n)) ;
+        //GrB_TRY (GrB_Vector_setElement(l_level, 0, src)) ;
+    }
+
+	// create a scalar to hold the destination value
+	GrB_Index dest_val ;
+
+    //--------------------------------------------------------------------------
+    // BFS traversal and label the nodes
+    //--------------------------------------------------------------------------
+    GrB_Index nq = 1 ;          // number of nodes in the current level
+    GrB_Index last_nq = 0 ;
+    GrB_Index current_level = 0;
+    GrB_Index nvals = 1;
+
+    // {!mask} is the set of unvisited nodes
+    GrB_Vector mask = (compute_parent) ? l_parent : l_level ;
+
+    // parent BFS
+    do
+    {
+        if (compute_level)
+        {
+            // assign levels: l_level<s(frontier)> = current_level
+            GrB_TRY( GrB_assign(l_level, frontier, GrB_NULL,
+                                current_level, GrB_ALL, n, GrB_DESC_S) );
+        }
+		++current_level;
+
+        if (compute_parent)
+        {
+            // frontier(i) currently contains the parent id of node i in tree.
+            // l_parent<s(frontier)> = frontier
+            GrB_TRY( GrB_assign(l_parent, frontier, GrB_NULL,
+                                frontier, GrB_ALL, n, GrB_DESC_S) );
+
+            // convert all stored values in frontier to their indices
+            // TODO: replace with apply(i) in GraphBLAS 2.0
+            GrB_TRY( GrB_eWiseMult(frontier, GrB_NULL, GrB_NULL, GrB_FIRST_UINT64,
```

Comment:
Use apply with either `GxB_POSITIONI_INT32` ` GxB_POSITIONI_INT64` no need to allocate `index_ramp`

Reply:
I tried making this change, but am not getting the correct results:
```
GrB_apply(frontier, GrB_NULL, GrB_NULL, unary_op, frontier, GrB_NULL) ;
```
The multiplication method had produced, for one of my sample graphs, the vector:
```
    (0,0)   0
    (1,0)   0
    (2,0)   1
    (3,0)   1
    (4,0)   0
```
The apply method produces:
```
    (0,0)   0                                                                  
    (1,0)   1                                                                                                                                                 
    (2,0)   2                                                                                                                                                 
    (3,0)   2                                                                  
    (4,0)   1 
```
Can you tell what I'm missing?